### ISSUE TITLE
#[bincode(bound)] for generic types

### ISSUE THREAD

**Issue by coolreader18** [bincode-org/bincode#511]
I'm looking into switching to bincode v2 and I'd love to get rid of serde - the only thing that seems to be lacking in the derive macros is that I can't specify custom bounds for generics. e.g.:

```rust
#[derive(Encode, Decode)]
struct A<B: MyTrait> {
    foo: usize,
    b: B,
    thing: B::Assoc,
}
```
With serde, I had a `#[serde(bound(deserialize = "B::Assoc: serde::Deserialize<'de>", serialize = "B::Assoc: serde::Serialize"))]`; it'd be awesome to have a similar feature for bincode v2's derive.

**Comment by coolreader18** [comment]
Oh, also, this is the generated code for a type with a default type param:
```rust
impl<C: Constant = ConstantData> ::bincode::Encode for CodeObject<C>
```
which is invalid syntax.

**Comment by VictorKoenders** [comment]
This is being implemented as a part of #526. We'll have to start splitting the `Decode` and `BorrowDecode` trait, which might take a while.
